sentry-ruby/lib/sentry/configuration.rb

```diff
@@ -435,19 +438,22 @@ def enabled_in_current_env?
       enabled_environments.empty? || enabled_environments.include?(environment)
     end
 
+    def valid_sample_rate?(sample_rate)
+      sample_rate &&
+      sample_rate.respond_to?(:>=) &&
+      sample_rate >= 0.0 &&
+      sample_rate.respond_to?(:<=) &&
+      sample_rate <= 1.0
```

Comment:
I think we can check the sample rate's type instead of the methods it supports. We don't need to be that flexible here.

```suggestion
      return false unless sample_rate.is_a?(Numeric)
      sample_rate >= 0.0 && sample_rate <= 1.0
```